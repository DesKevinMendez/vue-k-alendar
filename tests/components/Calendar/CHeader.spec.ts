
import CHeader from '@/components/Calendar/CHeader.vue';
import useConfig from '@/composables/useConfig';
import useRenderCalendar from '@/composables/useRenderCalendar';
import { mount, VueWrapper } from '@vue/test-utils';
import { DateTime } from 'luxon';
const { setLang } = useConfig()

let wrapper: VueWrapper<InstanceType<typeof CHeader>>

describe('CHeader', () => {

  beforeEach(() => {
    wrapper = mount(CHeader);
  });

  afterEach(() => {
    vi.clearAllMocks()
  })

  it('should render the component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should exists components', () => {
    expect(wrapper.findComponent({ name: 'ButtonPrevMonth' }).exists()).toBeTruthy();
    expect(wrapper.findComponent({ name: 'ButtonNextMonth' }).exists()).toBeTruthy();
    expect(wrapper.findComponent({ name: 'ButtonTodayMonth' }).exists()).toBeTruthy();
  })

  it('title come from useRenderCalendar', async () => {
    const { title } = useRenderCalendar()

    expect(wrapper.find('.center-title').text()).toBe(title.value)
  })

  describe('clicks on buttons', () => {

    afterEach(() => {
      vi.clearAllMocks()
    })

    it('click on ButtonPrevMonth', async () => {
      const { currentDate } = useRenderCalendar()

      currentDate.value = DateTime.fromISO('2024-10-24') as DateTime<true>

      const button = wrapper.findComponent({ name: 'ButtonPrevMonth' })

      await button.trigger('click')

      expect(wrapper.emitted('handlePrevMonth')).toStrictEqual([['2024-09-24']])
    })

    it('click on ButtonNextMonth', async () => {
      const { currentDate } = useRenderCalendar()

      currentDate.value = DateTime.fromISO('2024-10-24') as DateTime<true>

      const button = wrapper.findComponent({ name: 'ButtonNextMonth' })

      await button.trigger('click')

      expect(wrapper.emitted('handleNextMonth')).toStrictEqual([['2024-11-24']])
    })

    it('click on ButtonTodayMonth', async () => {
      const now = DateTime.now().toISODate()

      const button = wrapper.findComponent({ name: 'ButtonTodayMonth' })

      await button.trigger('click')

      expect(wrapper.emitted('handleToToday')).toStrictEqual([[now]])
    })
  })
});