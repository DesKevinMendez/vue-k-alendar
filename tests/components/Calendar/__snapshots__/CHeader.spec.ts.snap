// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CHeader > should render the component 1`] = `
"<header data-v-0a815d1e="" class="k-alendar-header-container">
  <div data-v-0a815d1e="" class="left-buttons">
    <div data-v-0a815d1e=""><button data-v-11092de0="" data-v-0a815d1e="" type="button" class="k-alendar-navegation-prev">←</button><button data-v-f3d09f65="" data-v-0a815d1e="" type="button" class="k-alendar-navegation-left">→</button></div><button data-v-8f6b2986="" data-v-0a815d1e="" type="button" class="k-alendar-today-button">Today</button>
  </div>
  <div data-v-0a815d1e="" class="center-title">
    <h2 data-v-0a815d1e="">October 2024</h2>
  </div>
</header>"
`;
